server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java

```diff
@@ -2391,6 +2399,32 @@ private Expression parseReplaceFunction(
     return functionExpression;
   }
 
+  private Expression parseSubStrFunction(
+      IoTDBSqlParser.ScalarFunctionExpressionContext subStrClause, boolean canUseFullPath) {
+    FunctionExpression functionExpression = new FunctionExpression(SUBSTRING_FUNCTION);
+    IoTDBSqlParser.SubStringExpressionContext subStringExpression =
+        subStrClause.subStringExpression();
+    functionExpression.addExpression(parseExpression(subStringExpression.input, canUseFullPath));
+    if (subStringExpression.startPosition != null) {
+      functionExpression.addAttribute(
+          SUBSTRING_START, parseStringLiteral(subStringExpression.startPosition.getText()));
```

Comment:
parseStringLiteral is not correct(when startPosition contais three or more digits).